models/user.js

```diff
@@ -20,7 +22,15 @@ module.exports = (sequelize, DataTypes) => {
       type: DataTypes.BOOLEAN,
       defaultValue: false
     },
-  });
+  }, {
+    hooks: {
+      beforeCreate: (user) => {
+        const salt = bcrypt.genSaltSync(10);
```

Comment:
This is good... However, when we need to authenticate the user, we'd need to check that the password is valid. To this, end, It might be useful to extract lines 28 and 29 into a method on the prototype of the `User` model/class and then create another method on the prototype for checking validity of the password.

Picture the prototype method as a method of a class that can be called on all objects created from the class. This is why we are hashing the password on the User model in the first place so that all users created have access to those methods.

TLDR, what you'd do will bear semblance to:

      ```hooks: {
        beforeCreate(user) {
          user.hashPassword();
        }
      }```

And then Above `return User` in the model, you have these two functions. I put dummies below:

  ```User.prototype.hashPassword = function hashPassword() {
    .......;
  };```

  ```User.prototype.validPassword = function validPassword(password) {
    return ....;
  };```

Reply:
Check the `to hash a password` and `to check a password` sections of the `bcrypt` npm module to see what could be the contents of the two prototype methods above.

Reply:
I will work on it